### DISCUSSION TITLE
Do not navigate unless changes in dialog or panel is saved

### DISCUSSION THREAD

**Discussion by TechAspirant** [serenity-is/Serenity#5856] [Help]
Hello,

Is there any way I can restrict user to navigate to some other page (Eg. From Orders to Customer) unless user saved his changes in a dialog or panel If any ?

**Comment by minhhungit** [comment]
use permission on navigation
and when user do save, update permission table

**Reply by TechAspirant** [reply]
Any sample code ?

**Reply by go2ismail** [reply]
Hi,
You can use pendingChangeConfirmation at your dialog
`private loadedState: string;

        constructor() {
            super();
            Serenity.Extensions.DialogUtils.pendingChangesConfirmation(this.element, () => this.getSaveState() != this.loadedState);
            
        }

        

        getSaveState() {
            try {
                return $.toJSON(this.getSaveEntity());
            }
            catch (e) {
                return null;
            }
        }

        loadResponse(data) {
            super.loadResponse(data);
            this.loadedState = this.getSaveState();
        }`
